### PR TITLE
SAK-42763 Bubble attachment exception in mailsender

### PR DESCRIPTION
**Pull request by adrianfish**
https://jira.sakaiproject.org/browse/SAK-42763

### CODE REVIEW

File: mailsender/tool/src/webapp/WEB-INF/bundle/Messages.properties [hunk 1]
```diff
@@ -58,6 +58,7 @@ exception.generic=A problem has occurred: {0}
 error.unknown.role.type=An unknown role type was requested.
 error.no.recipients=No recipients specified.
 error.no.valid.recipients=No valid recipients specified.
+error.attachmentsize.exception=You added too many attachments. Limit: {0} MB
```
Comment (fostersdesign) [inline mailsender/tool/src/webapp/WEB-INF/bundle/Messages.properties]: Is the number of attachments what causes the "-1"?

"too many attachments" and a size don't seem to be compatible units.

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 16
           

Complexity increasing per file
==============================
- kernel/api/src/main/java/org/sakaiproject/email/api/AttachmentSizeException.java  1
- mailsender/impl/src/java/org/sakaiproject/mailsender/logic/impl/ExternalLogicImpl.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/sakaiproject/sakai/pullRequest?prid=4710155&bid=15732701)

File: mailsender/tool/src/webapp/WEB-INF/bundle/Messages.properties (hunk 1 above)
Comment (adrianfish) [reply mailsender/tool/src/webapp/WEB-INF/bundle/Messages.properties]: No, it's just the way the java library reports size. -1 is to be expected, unless you dig in deeper and get the underlying file size, which is what I've done.

Comment (adrianfish) [reply mailsender/tool/src/webapp/WEB-INF/bundle/Messages.properties]: Ah, I get you. The message is a bit odd.

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 16
           

Complexity increasing per file
==============================
- kernel/api/src/main/java/org/sakaiproject/email/api/AttachmentSizeException.java  1
- mailsender/impl/src/java/org/sakaiproject/mailsender/logic/impl/ExternalLogicImpl.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/sakaiproject/sakai/pullRequest?prid=4710155&bid=15732701)

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- mailsender/impl/src/java/org/sakaiproject/mailsender/logic/impl/ExternalLogicImpl.java  2
- kernel/api/src/main/java/org/sakaiproject/email/api/AttachmentSizeException.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/sakaiproject/sakai/pullRequest?prid=4710155&bid=15732701)

File: kernel/api/src/main/java/org/sakaiproject/email/api/Attachment.java [hunk 2]
```diff
@@ -149,6 +150,18 @@ public String getContentDispositionHeader()
 		return contentDisposition;
 	}
 
+	public Optional<Long> getSizeIfFile() {
```
Comment (ern) [inline kernel/api/src/main/java/org/sakaiproject/email/api/Attachment.java]: does getSizeOfFile read better?

File: kernel/api/src/main/java/org/sakaiproject/email/api/AttachmentSizeException.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ **********************************************************************************/
+package org.sakaiproject.email.api;
+
+public class AttachmentSizeException extends Exception {
```
Comment (ern) [inline kernel/api/src/main/java/org/sakaiproject/email/api/AttachmentSizeException.java]: Should the Exception reference the size some how?

File: kernel/kernel-impl/src/main/java/org/sakaiproject/email/impl/BasicEmailService.java [hunk 4]
```diff
@@ -376,35 +232,35 @@ public void init()
...
-				+ " maxRecipients: " + m_maxRecipients + " testMode: " + m_testMode
-				+ ((m_smtpConnectionTimeout != null) ? (" smtpConnectionTimeout: " + m_smtpConnectionTimeout) : "")
-				+ ((m_smtpTimeout != null) ? (" smtpTimeout: " + m_smtpTimeout) : ""));
+		if (smtp != null) System.setProperty(propName(MAIL_HOST_T), smtp);
+		if (smtpPort != null) System.setProperty(propName(MAIL_PORT_T), smtpPort);
+		System.setProperty(propName(MAIL_FROM_T), smtpFrom);
+		if (smtpConnectionTimeout != null) System.setProperty(propName(MAIL_CONNECTIONTIMEOUT_T), smtpConnectionTimeout);
+		if (smtpTimeout != null) System.setProperty(propName(MAIL_TIMEOUT_T), smtpTimeout);
+
+		log.info("init(): smtp: " + smtp + ((smtpPort != null) ? (":" + smtpPort) : "") + " bounces to: " + smtpFrom
```
Comment (ern) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/email/impl/BasicEmailService.java]: log statement could really benefit from the newer style!

File: kernel/api/src/main/java/org/sakaiproject/email/api/Attachment.java (hunk 2 above)
Comment (adrianfish) [reply kernel/api/src/main/java/org/sakaiproject/email/api/Attachment.java]: It might not be a file, that's why I named it like that.

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- mailsender/impl/src/java/org/sakaiproject/mailsender/logic/impl/ExternalLogicImpl.java  2
- kernel/api/src/main/java/org/sakaiproject/email/api/AttachmentSizeException.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/sakaiproject/sakai/pullRequest?prid=4710155&bid=15732701)

File: kernel/api/src/main/java/org/sakaiproject/email/api/AttachmentSizeException.java (hunk 3 above)
Comment (adrianfish) [reply kernel/api/src/main/java/org/sakaiproject/email/api/AttachmentSizeException.java]: Added. New screenshot added to the jira.

**Pull request merged**